packages/util/src/emulator.ts

```diff
@@ -140,3 +141,153 @@ export function createMockUserToken(
     signature
   ].join('.');
 }
+
+interface EmulatorStatusMap {
+  [name: string]: boolean;
+}
+const emulatorStatus: EmulatorStatusMap = {};
+
+interface EmulatorSummary {
+  prod: string[];
+  emulator: string[];
+}
+
+// Checks whether any products are running on an emulator
+function getEmulatorSummary(): EmulatorSummary {
+  const summary: EmulatorSummary = {
+    prod: [],
+    emulator: []
+  };
+  for (const key of Object.keys(emulatorStatus)) {
+    if (emulatorStatus[key]) {
+      summary.emulator.push(key);
+    } else {
+      summary.prod.push(key);
+    }
+  }
+  return summary;
+}
+
+function getOrCreateEl(id: string): { created: boolean; element: HTMLElement } {
+  let parentDiv = document.getElementById(id);
+  let created = false;
+  if (!parentDiv) {
+    parentDiv = document.createElement('div');
+    parentDiv.setAttribute('id', id);
+    created = true;
+  }
+  return { created, element: parentDiv };
+}
+
+/**
+ * Updates Emulator Banner. Primarily used for Firebase Studio
+ * @param name
+ * @param isRunningEmulator
+ * @public
+ */
+export function updateEmulatorBanner(
+  name: string,
+  isRunningEmulator: boolean
+): void {
+  if (
+    typeof window === 'undefined' ||
+    typeof document === 'undefined' ||
+    !isCloudWorkstation(window.location.host) ||
+    emulatorStatus[name] === isRunningEmulator
```

Comment:
add a check here if it's been previously dismissed, so it doesn't just pop back up

Reply:
Already done.